pandas/io/parsers.py

```diff
@@ -757,6 +757,8 @@ def read_table(
     memory_map=False,
     float_precision=None,
 ):
+    if delim_whitespace:
+        sep = ","
```

Comment:
This needs a check that the user did not pass in `sep` different than `'\t'` ?

Reply:
Yes, you are right of course. Thanks

Reply:
Unlike 1.0.5 we don't raise on `pd.read_table(buf, sep=",", delim_whitespace=True)`.

I think it's unfortunate but will need to duplicate the 

```

    if delim_whitespace and delimiter != default_sep:
        raise ValueError(
            "Specified a delimiter with both sep and "
            "delim_whitespace=True; you can only specify one."
        )
```
here and add test to ensure raising like before.

Reply:
What do you think about creating a function _read_csv which is called from read_csv and read_table where default_sep is an additional Input? That would avoid the if clause and the duplication

Reply:
I think if planning a follow-up, can deduplicate then instead of making too many changes in a backport candidate. Again a code comment along the lines `TODO: validation duplicated in read_csv` might help.

Reply:
Thanks, done.